src/hotspot/share/nmt/regionsTree.hpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+#ifndef NMT_REGIONSTREE_HPP
+#define NMT_REGIONSTREE_HPP
+
+#include "nmt/nmtCommon.hpp"
+#include "nmt/virtualMemoryTrackerCommon.hpp"
+#include "nmt/vmatree.hpp"
+
+class RegionsTree : public VMATree {
+ private:
+  NativeCallStackStorage* _ncs_storage;
```

Comment:
No need for this to be a pointer.

Reply:
How to use the ctor with a `bool` arg, then?

Reply:
Done.
For my curiosity, what is the advantage?

Reply:
1. No malloc
2. No indirection, so no cache misses
3. A clear lifetime and clear ownership, both are bound to the `RegionsTree` object

Reply:
OK. Thanks for your description.